bundles/binding/org.openhab.binding.mystromecopower/src/main/java/org/openhab/binding/mystromecopower/internal/api/MystromClient.java

```diff
@@ -0,0 +1,304 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.mystromecopower.internal.api;
+
+import org.openhab.binding.mystromecopower.internal.api.model.*;
+import org.slf4j.Logger;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.util.ArrayList;
+import java.util.List;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * Manage Json Api call to mystrom Api.
+ * @since 1.8.0-SNAPSHOT
+ * @author Jordens Christophe
+ *
+ */
+public class MystromClient implements IMystromClient{
+	private Logger logger;
+	private JsonParser jsonParser;
+	private Gson gson;
+	private String userName;
+	private String password;
+	private String authToken;
+	private static final String API_URL = "https://mystrom.ch/mobile/";
+	
+	/**
+	 * Initialize the MystromClient class.
+	 * @param userName The user name for the mystrom server connection.
+	 * @param password The password for the mystrom server connection.
+	 * @param logger The logger used to log into openhab.
+	 */
+	public MystromClient(String userName, String password, Logger logger)
+	{
+		this.gson = this.createGsonBuilder().create();
+		this.logger = logger;
+		this.userName = userName;
+		this.password = password;
+		this.jsonParser = new JsonParser();
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.openhab.binding.mystromecopower.internal.api.IMystromClient#login()
+	 */
+	public Boolean login(){
+		Reader reader = null;
+		logger.debug("Do login for user '{}'", this.userName);
+		
+		try {
+			HttpURLConnection httpURLConnection;
+			httpURLConnection = (HttpURLConnection) new URL(API_URL + "auth?email="
+					+ this.userName
+					+ "&password=" + this.password)
+					.openConnection();
+		
+			InputStream inputStream = httpURLConnection.getInputStream();
+			reader = new InputStreamReader(inputStream, "UTF-8");
+			JsonObject jsonObject = (JsonObject) jsonParser.parse(reader);
+			
+			String status = jsonObject.get("status").getAsString();
+			if(!status.equals("ok")){
+				return false;
+			}
+			logger.debug("Logon successfull");
+			
+			authToken = jsonObject.get("authToken").getAsString();
+			
+			return true;
+		}
+		catch(Exception ex)
+		{
+			logger.error("Error do logon: '{}'", ex.toString());
+			return false;
+		} 
+		finally {
+			if (reader != null) {
+				try {
+					reader.close();
+				} catch (IOException ignored) {
+
+				}
+			}
+		}
+	}
+	
+	/* (non-Javadoc)
+	 * @see org.openhab.binding.mystromecopower.internal.api.IMystromClient#getDevices()
+	 */
+	public List<MystromDevice> getDevices()
+	{
+		Reader reader = null;
+		
+		logger.info("get devices...");
+		
+		try {
+			HttpURLConnection httpURLConnection;
+			httpURLConnection = (HttpURLConnection) new URL(API_URL + "devices" +
```

Comment:
Could you please assign the URL to a local variable before opening the connection? This helps while debugging problems.